File: src/main/java/net/starlegacy/feature/gas/collectionfactors/SkyLightFactor.java
package net.starlegacy.feature.gas.collectionfactors;

import org.bukkit.Location;

public class SkyLightFactor extends CollectionFactor {
    private int minimum;
    private int maximum;

    public SkyLightFactor(int minimum, int maximum) {
        this.minimum = minimum;
        this.maximum = maximum;
    }

    @Override
    public boolean factor(Location location) {
        return location.getBlock().getLightFromSky() <= maximum && location.getBlock().getLightFromSky() >= minimum;
    }
}

File: server/src/main/kotlin/net/horizonsend/ion/server/features/gas/collectionfactors/SkyLightFactor.kt
package net.horizonsend.ion.server.features.gas.collectionfactors

import org.bukkit.Location

class SkyLightFactor(private val minimum: Int, private val maximum: Int) : CollectionFactor() {
    override fun factor(location: Location): Boolean {
        return location.block.lightFromSky in minimum..maximum
    }

	override fun canBeFound(location: Location): Boolean = factor(location)
}
